woningwaardering/stelsels/zelfstandige_woonruimten/keuken/keuken_2024.py

```diff
@@ -0,0 +1,158 @@
+from decimal import ROUND_HALF_UP, Decimal
+
+from loguru import logger
+
+
+from woningwaardering.stelsels.stelselgroepversie import Stelselgroepversie
+from woningwaardering.stelsels.utils import naar_tabel
+from woningwaardering.vera.bvg.generated import (
+    EenhedenEenheid,
+    WoningwaarderingResultatenWoningwaardering,
+    WoningwaarderingResultatenWoningwaarderingCriterium,
+    WoningwaarderingResultatenWoningwaarderingCriteriumGroep,
+    WoningwaarderingResultatenWoningwaarderingGroep,
+    WoningwaarderingResultatenWoningwaarderingResultaat,
+)
+from woningwaardering.vera.referentiedata import (
+    Woningwaarderingstelsel,
+    Woningwaarderingstelselgroep,
+)
+from woningwaardering.vera.referentiedata.bouwkundigelementdetailsoort import (
+    Bouwkundigelementdetailsoort,
+)
+from woningwaardering.vera.referentiedata.meeteenheid import Meeteenheid
+from woningwaardering.vera.referentiedata.ruimtedetailsoort import Ruimtedetailsoort
+
+
+class Keuken2024(Stelselgroepversie):
+    @staticmethod
+    def bereken(
+        eenheid: EenhedenEenheid,
+        woningwaardering_resultaat: (
+            WoningwaarderingResultatenWoningwaarderingResultaat | None
+        ) = None,
+    ) -> WoningwaarderingResultatenWoningwaarderingGroep:
+        woningwaardering_groep = WoningwaarderingResultatenWoningwaarderingGroep(
+            criteriumGroep=WoningwaarderingResultatenWoningwaarderingCriteriumGroep(
+                stelsel=Woningwaarderingstelsel.zelfstandige_woonruimten.value,
+                stelselgroep=Woningwaarderingstelselgroep.keuken.value,
+            )
+        )
+        woningwaardering_groep.woningwaarderingen = []
+
+        geldige_ruimtes = [
+            ruimte
+            for ruimte in eenheid.ruimten or []
+            # check of een ruimte behoort tot te ruimtesoorten die in aanmerking komen voor de stelselgroep keuken
+            if ruimte.detail_soort
+            and (
+                ruimte.detail_soort.code == Ruimtedetailsoort.keuken.code
+                or ruimte.detail_soort.code
+                == Ruimtedetailsoort.woonkamer_en_of_keuken.code
+                or ruimte.detail_soort.code == Ruimtedetailsoort.woonkamer.code
+                or ruimte.detail_soort.code
+                == Ruimtedetailsoort.woon_en_of_slaapkamer.code
+                or ruimte.detail_soort.code == Ruimtedetailsoort.slaapkamer.code
+            )
```

Comment:
Misschien nog zoiets voor de leesbaarheid:

```suggestion
            and (
                ruimte.detail_soort.code
                in [
                    Ruimtedetailsoort.keuken.code,
                    Ruimtedetailsoort.woonkamer_en_of_keuken.code,
                    Ruimtedetailsoort.woonkamer.code,
                    Ruimtedetailsoort.woon_en_of_slaapkamer.code,
                    Ruimtedetailsoort.slaapkamer.code,
                ]
            )
```